src/org/labkey/test/util/exp/DataClassAPIHelper.java

```diff
@@ -51,4 +60,31 @@ public static List<FieldDefinition> dataClassTestFields()
                 new FieldDefinition("attachmentColumn", FieldDefinition.ColumnType.Attachment));
     }
 
+    public static Map<String, Integer> getRowIdsForDataEntity(String containerPath, String dataTypeName, List<String> entityNames) throws IOException, CommandException
+    {
+
+        Connection connection = WebTestHelper.getRemoteApiConnection();
+        SelectRowsCommand cmd = new SelectRowsCommand("exp.data", dataTypeName);
+        cmd.setColumns(Arrays.asList("RowId", "Name"));
+        cmd.addFilter("Name", String.join(";", entityNames), Filter.Operator.IN);
+
+        SelectRowsResponse response = cmd.execute(connection, containerPath);
+
+        Map<String, Integer> rowIds = new HashMap<>();
+
+        for(Map<String, Object> row : response.getRows())
+        {
+            Object name = row.get("Name");
+            Object value = row.get("RowId");
+            rowIds.put(name.toString(), Integer.parseInt(value.toString()));
+        }
+
+        // Check that the names returned from the query match the names sent in.
+        Set<String> names = new HashSet<>(entityNames);
+        Assert.assertTrue("The names of the data entities returned from the query do not match the names sent in.",
```

Comment:
Should they be an exact match? Or is one a superset of the other? If they're the same, doing an Assert.assertEquals() has the nice benefit of giving more detail about their contents when the check fails.

Reply:
Good point, they should be  identical. Changed the check to assertEquals.